src/tools/ci/test_pkg_conf

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import shutil
+import subprocess
+import os
+import sys
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--rootdir', help='root directory', type=str, required=True)
+parser.add_argument('--pkgconf_installdir', help='pkgconf install directory', type=str, required=True)
+parser.add_argument('--builddir', help='build directory', type=str, required=True)
+parser.add_argument('--meson', help='path to meson binary', type=str, required=True)
+
+args = parser.parse_args()
+
+rootdir = os.path.realpath(args.rootdir)
+builddir = os.path.realpath(args.builddir)
+pkgconf_installdir = os.path.realpath(args.pkgconf_installdir)
+adminpackdir = os.path.join(rootdir, 'contrib/adminpack')
+workdir = os.path.join(builddir, 'contrib/adminpack_tmp')
+exit_code = 0
+
+adminpack_meson_build_file = \
+  '''
+  project('adminpack', 'c')
+
+  pg_ext = dependency('postgresql-extension-warnings')
+
+  adminpack = shared_module('adminpack',
+    ['adminpack.c'],
+    dependencies: pg_ext,
+    name_prefix: '',
+    install_dir: pg_ext.get_variable(pkgconfig: 'dir_mod')
+  )
+
+  install_data(
+    'adminpack.control',
+    'adminpack--1.0.sql',
+    'adminpack--1.0--1.1.sql',
+    'adminpack--1.1--2.0.sql',
+    'adminpack--2.0--2.1.sql',
+    install_dir: pg_ext.get_variable(pkgconfig: 'dir_data')
+  )
+  '''
+
+# copy adminpack
+if os.path.exists(workdir):
+    shutil.rmtree(workdir)
+shutil.copytree(adminpackdir, workdir)
+
+# overwrite meson.build file
+meson_file = os.path.join(workdir, 'meson.build')
+with open(meson_file, 'w') as f:
+    f.write(adminpack_meson_build_file)
+
+
+def sp_check(sp):
+    global exit_code
+    if sp.returncode != 0:
+        exit_code = sp.returncode
+    return sp.returncode
+
+
+# run tests
+def run_tests(pkg_conf_path, message=''):
+    global exit_code
```

Comment:
Ick, do we really need this? Seems it'd be cleaner to count the number of failed run_tests() invocations? The sp_check() invocations can just be something like

```python
    if subprocess.run(meson_setup_command, env=env, cwd=workdir).returncode != 0:
        return False
    if subprocess.run(ninja_build_command, cwd=workdir).returncode != 0:
        return False
    return True
```

Reply:
I don't think this is quite resolved - the global is still there...

Reply:
Ping?